### PR TITLE
Add index and standard for charm configuration option description

### PR DESCRIPTION
**Pull request by amandahla**
- Add index
- Add a standard for charm configuration option description (">" or "|" for description and its impact on Charmhub charm page).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -81,6 +96,64 @@ the `is-charms` team as reviewer
 *       @canonical/is-charms
 ```
 
+## Charm Configuration Option Description
+
+The charm configuration is defined in a single file named config.yaml. Each configuration option can define a `description` that contains an explanation of the configuration item. When a charm is published to the official repository of charms, [Charmhub](https://charmhub.io/), this configuration file will be used as content for the `Configure` page.
```
Comment (jdkandersson) [inline CONTRIBUTING.md:101]: The other lines in this file are limited to 80 characters to make it easy to read, could these please be updated to that as well?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -81,6 +96,64 @@ the `is-charms` team as reviewer
...
+The Block Scalar format has three parts: Block Style Indicator, Block Chomping Indicator and Indentation Indicator. More information in [YAML Multiline](https://yaml-multiline.info/).
+
+The Block Style Indicator indicates how newlines inside the block should behave.
+
+- `>`  new lines will be replaced by spaces.
+- `|`  new lines will be kept as newlines.
+
+The choice affects how the Charmhub documentation is presented to the user so it's important to format the description and choose the indicator accordingly.
+
+Real example: [Alertmanager k8s Charm](https://charmhub.io/alertmanager-k8s).
```
Comment (jdkandersson) [inline CONTRIBUTING.md:114]: The example could be updated in the future, perhaps a screenshot/ text based example for both on charmhub would help?

**Review by jdkandersson — COMMENTED** [review]
Thank you for this!

File: CONTRIBUTING.md (hunk 2 above)
Comment (amandahla) [reply CONTRIBUTING.md:181]: For both, do you mean the Block Style Indicators? The YAML excerpt from Alermanager K8S shows both.

Comment (jdkandersson) [reply CONTRIBUTING.md:181]: Yes I think that is sufficient

**Review by jdkandersson — APPROVED** [review]

**Review by arturo-seijas — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -138,6 +153,75 @@ the `is-charms` team as reviewer
...
+
+      This should be a complete URI, including scheme, or a  fully qualified subpath
+      starting with `/`.
+      If Alertmanager is being served directly from the root of a fully-qualified
+      host or a bare A record, this may be omitted.
+      If the URL has a path portion, Alertmanager will use it to prefix all HTTP
+      endpoints.
+```
+
+And this is how is shown in its Charmhub [page](https://charmhub.io/alertmanager-k8s/configure):
```
Comment (gregory-schiano) [inline CONTRIBUTING.md:221]: I would rather write:
`And this is how it's shown in Charmhub` to avoid "is, is, in, its" which is quite a gymnastic :p

Comment (amandahla) [reply CONTRIBUTING.md:221]: You're right, I didn't notice that. Almost like a rap song :notes: :smiley: I'll change it in the next PR.

**Pull request merged**